geomstats/geometry/dirichlet_distributions.py

```diff
@@ -441,16 +560,39 @@ def initialize(point_0, point_1):
                 def bc(y0, y1, ip=ip, ep=ep):
                     return boundary_cond(y0, y1, ip, ep)
 
-                solution = solve_bvp(bvp, bc, t_int, geodesic_init)
-                solution_at_t = solution.sol(t)
-                geodesic = solution_at_t[:self.dim, :]
-                geod.append(gs.squeeze(gs.transpose(geodesic)))
+                def process_function(return_dict):
+                    if jacobian:
+                        solution = solve_bvp(
+                            bvp, bc, t_int, geodesic_init, fun_jac=jac)
+                    else:
+                        solution = solve_bvp(bvp, bc, t_int, geodesic_init)
+                    solution_at_t = solution.sol(t)
+                    geodesic = solution_at_t[:self.dim, :]
+                    geod.append(gs.squeeze(gs.transpose(geodesic)))
+                    return_dict[0] = geod
+
+                manager = multiprocessing.Manager()
+                return_dict = manager.dict()
+                process = multiprocessing.Process(
+                    target=process_function, args=(return_dict,))
+                process.start()
+
+                process.join(max_time)
+                if process.is_alive():
+                    process.terminate()
+                    print('Maximum time of {} seconds reached. '
+                          'Process terminated. '
+                          'Result is inaccurate.'.format(max_time))
+                    geod.append(gs.zeros((n_steps, self.dim)))
```

Comment:
Use logging.info instead of print

Reply:
Maybe returning NaNs would be better than zeros?

Reply:
You're right.